src/prpy/serialization.py

```diff
@@ -320,44 +354,83 @@ def deserialize_environment(data, env=None, purge=False, reuse_bodies=None):
-        for joint_data in data['joints']
-    ]
-
-    if data['is_robot']:
-        # TODO: Also load sensors.
-        manipulator_infos = [
-            deserialize_manipulator_info(manipulator_data['info']) \
-            for manipulator_data in data['manipulators']
-        ]
-        sensor_infos = []
-
-        kinbody = RaveCreateRobot(env, '')
-        kinbody.Init(
-            link_infos, joint_infos,
-            manipulator_infos, sensor_infos,
-            data['uri']
-        )
+    
+    name_desired = name or data['name']
+    
+    if data.get('uri_only', False):
+        
+        if data['is_robot']:
+
+            parts = data['uri'].split()
+            if len(parts)==2 and parts[0].endswith('.urdf') and parts[1].endswith('.srdf'):
+                robot_name = urdf_module_getter().SendCommand('Load {}'.format(data['uri']))
+                if robot_name != name_desired:
+                    raise RuntimeError('error, or_urdf name mismatch!')
+                kinbody = env.GetRobot(robot_name)
+            else:
+                kinbody = env.ReadRobotXMLFile(data['uri'])
+                kinbody.SetName(name_desired)
+                env.Add(kinbody, anonymous)
+            
+        else:
+
+            if data['uri'].endswith('.urdf'):
+                kinbody_name = urdf_module_getter().SendCommand('Load {}'.format(data['uri']))
+                if kinbody_name != name_desired:
+                    raise RuntimeError('error, or_urdf name mismatch!')
+                kinbody = env.GetKinBody(kinbody_name)
+            else:
+                kinbody = env.ReadKinBodyXMLFile(data['uri'])
+                kinbody.SetName(name_desired)
+                env.Add(kinbody)
+        
     else:
-        kinbody = RaveCreateKinBody(env, '')
-        kinbody.Init(link_infos, joint_infos, data['uri'])
-
-    kinbody.SetName(name or data['name'])
-    env.Add(kinbody, anonymous)
+        
+        link_infos = [
+            deserialize_link_info(link_data['info']) \
+            for link_data in data['links']
+        ]
+        joint_infos = [
+            deserialize_joint_info(joint_data['info']) \
+            for joint_data in data['joints']
+        ]
 
+        if data['is_robot']:
+            
+            # TODO: Also load sensors.
+            manipulator_infos = [
+                deserialize_manipulator_info(manipulator_data['info']) \
+                for manipulator_data in data['manipulators']
+            ]
+            sensor_infos = []
+        
+            kinbody = RaveCreateRobot(env, '')
+            kinbody.Init(
+                link_infos, joint_infos,
+                manipulator_infos, sensor_infos,
+                data['uri']
+            )
+        
+        else:
+            
+            kinbody = RaveCreateKinBody(env, '')
+            kinbody.Init(link_infos, joint_infos, data['uri'])
+        
+        kinbody.SetName(name_desired)
+        env.Add(kinbody, anonymous)
```

Comment:
I looked at this, and I think it's correct ... there's an `else:` above, so this code should only be executed if `uri_only` is not set.